### PR TITLE
QGB Relayer add commitments check when relaying confirms

### PR DESCRIPTION
**Pull request by SweeXordious**
Adds a check on the confirms' commitments before relaying them

### CODE REVIEW

**Review by adlerjohn — APPROVED** [review]
utACK

**Review by evan-forbes — APPROVED** [review]
:+1: :+1:

**Comment by SweeXordious** [comment]
Will not merge until I get to the bottom of the commitments issue.

File: x/qgb/orchestrator/relayer.go [hunk 1]
```diff
@@ -147,6 +147,22 @@ func (r *Relayer) submitDataRootTupleRoot(
...
+			r.logger.Error("confirms don't have the same commitment",
+				"expected_commitment",
+				commitment,
+				"confirm_commitment",
+				conf.Commitment,
+				"nonce",
+				conf.Nonce,
+				"signer_eth_address",
+				conf.EthAddress)
+			return ErrInvalidCommitmentInConfirm
```
Comment (SweeXordious) [inline x/qgb/orchestrator/relayer.go:162]: @evan-forbes What do you think makes more sense:
- not relay attestations where one or more data commitments have an invalid commitment?
- relay if we have enough data commitments that are good (without considering the faulty ones).

I guess we can go for second option for now. 
What do you think?

Comment (evan-forbes) [reply x/qgb/orchestrator/relayer.go:162]: Agree that option 2 is probably best. We want to ignore potentially malicious validators. In the future, we will slash for this, as they sign an incorrect data commitment.

Comment (SweeXordious) [reply x/qgb/orchestrator/relayer.go:162]: Awesome. Thanks a lot.

**Comment by SweeXordious** [comment]
We could even increase the validations we do on the commitments before adding them to make the relayer smarter and not just keep sending transactions to the EVM chain that we know won't pass.
What do you think? Should we create an issue for that?

**Comment by evan-forbes** [comment]
> We could even increase the validations we do on the commitments before adding them to make the relayer smarter and not just keep sending transactions to the EVM chain that we know won't pass.
What do you think? Should we create an issue for that?

hey sorry I missed this. Yeah, this sounds good! Perhaps this should be covered in ADR 005?

**Pull request merged**